imageregistry/v1/types.go

```diff
@@ -303,8 +302,65 @@ type ImageRegistryConfigStorageAzure struct {
 	// object.
 	// +optional
 	CloudName string `json:"cloudName,omitempty"`
+	// networkAccess defines the network access properties for the storage account.
+	// Defaults to type: External.
+	// +kubebuilder:default={"type": "External"}
+	// +optional
+	NetworkAccess *AzureNetworkAccess `json:"networkAccess,omitempty"`
+}
+
+type AzureNetworkAccess struct {
```

Comment:
This is a discriminated union. Should set up the correct tags and CEL based on the example in the `examples/v1` API. You seem to have an optional member so have a look at the example of an optional DU member there

Reply:
Done. PTAL